src/diffpy/labpdfproc/tools.py

```diff
@@ -180,6 +182,49 @@ def set_mud(args):
     return args
 
 
+def set_mud_using_xraydb(args):
+    """
+    Set the mud using xraydb, prompting for parameters if not provided by the user
+
+    The function works in this way:
+    (1) if mu*D is provided, no further input is required, and any additional inputs are written to metadata.
+    (2) if mu*D is missing but mu is provided, prompt for diameter if missing and compute muD = mu * d.
+    (3) if neither is provided, prompt for sample name, energy, density, and diameter as needed, then compute muD.
+    Reference for the xraydb database: https://xraypy.github.io/XrayDB/python.html#xraydb.material_mu.
+
+    Parameters
+    ----------
+    args argparse.Namespace
+        the arguments from the parser
+    sample str
+        the chemical formula or name of material
+    energy float
+        the energy in eV
+    density float or None
+        material density in gr/cm^3
+    diameter float
+        the capillary diameter in mm
+
+    Returns
+    -------
+    the updated args argparse.Namespace with mu*D
+    """
+
+    if args.mud:
+        return args
+    if args.mu:
+        args.diameter = args.diameter or float(input("Please enter the capillary diameter (mm): ").strip())
+        args.mud = args.mu * args.diameter
+        return args
+    args.sample = args.sample or input("Please enter the chemical formula or name of material: ").strip()
```

Comment:
I added some input prompts because it might be easier to enter from here than from cli inputs.